### ISSUE TITLE
CMake's find_library doesn't search outside the default paths

### ISSUE THREAD

**Issue by rkfg** [mxe/mxe#2018]
I'm building a simple project that uses [C++ Actor Framework](https://github.com/Actor-framework/actor-framework/). I've successfully built the framework itself (after some minor patches like Windows.h => windows.h and such) and installed the libs and headers to `/opt/build/win64`. The problem is that mxe's CMake behaves differently compared to the system one. I have the same framework compiled for Linux x86_64, installed to `/opt/build/static/lib`. I can easily find the libraries with this:
```
find_library(CAFCORE NAMES caf_core_static caf_core)
find_library(CAFIO NAMES caf_io_static caf_io)
```
using `cmake -DCMAKE_LIBRARY_PATH=/opt/build/static/lib ..` (the system one). However, mxe's CMake can't find the libs when I do:
```
~/soft/mxe_latest/usr/bin/x86_64-w64-mingw32.static.posix-cmake -DCMAKE_LIBRARY_PATH=/opt/build/win64/lib ..

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CAFCORE
    linked by target "pg" in directory /home/rkfg/cpp/pg
CAFIO
    linked by target "pg" in directory /home/rkfg/cpp/pg

-- Configuring incomplete, errors occurred!
```

I tried many other CMake vars that are described in [the manual](https://cmake.org/cmake/help/v3.10/command/find_library.html) but they just don't work. The only way is to symlink those libs to `~/soft/mxe_latest/usr/x86_64-w64-mingw32.static.posix/lib`, then CMake finds and uses them just fine. But that's an obviously bad solution. Paths, filenames, prefixes, suffixes, invisible unicode characters between letters etc. are doublechecked. Tested against today's master branch, cloned and built from scratch.

**Comment by rkfg** [comment]
Yes, there's also an option to just `make install` the program into mxe root but it would leave a lot of trash behind. `make install` is generally not recommended and `checkinstall` won't work with mxe. I need to be able to keep those libs in some external location and supply them when needed.

**Comment by rkfg** [comment]
So I fixed this for me with this:
```diff
diff --git a/src/cmake/conf/mxe-conf.cmake.in b/src/cmake/conf/mxe-conf.cmake.in
index 8bd4cf1f..4c6d625f 100644
--- a/src/cmake/conf/mxe-conf.cmake.in
+++ b/src/cmake/conf/mxe-conf.cmake.in
@@ -41,8 +41,8 @@ set(LIBTYPE @LIBTYPE@)
 ## Paths etc.
 set(CMAKE_FIND_ROOT_PATH @PREFIX@/@TARGET@)
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
-set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
 set(CMAKE_PREFIX_PATH @PREFIX@/@TARGET@)
 set(CMAKE_INSTALL_PREFIX @PREFIX@/@TARGET@ CACHE PATH "Installation Prefix")
 # For custom mxe FindPackage scripts
```
Now I can use the `CMAKE_LIBRARY_PATH` option. What are the reasons behind the `ONLY` value? Does it prevent clashing with the host system libraries?

**Comment by zavodnyrichard** [comment]
I'm having similiar problem. When I use **cmake**, everything is fine, but when I use **i686-w64-mingw32.static-cmake** (MXE wrapper), then I can't find my library which sit in /usr/local directory. 

I looking for lib this way:
`find_library(CryptoPP cryptopp)`

And getting this error:
`CryptoPP
    linked by target "cryptotest" in directory /run/media/zavodnyrichard/data/projects/cryptotest`

**Comment by zavodnyrichard** [comment]
Does anybody know why? ^^

**Comment by rkfg** [comment]
I provided a fix, MXE seems deliberately not to look outside its own root path. You can change it and do `make cmake-conf`.

**Comment by zavodnyrichard** [comment]
So change the two lines from ONLY to BOTH and rebuild cmake-conf package?

**Comment by rkfg** [comment]
Yes, that should do it.

**Comment by zavodnyrichard** [comment]
Nope. Still getting the same errors. Even if I use the `-DCMAKE_LIBRARY_PATH` option and point it to /usr/local, where my library sits.

**Comment by rkfg** [comment]
Try to touch `src/cmake.mk` and rebuild cmake as well. Also, you probably shouldn't mix your host libraries that are usually located in `/usr` and crossplatform libs. I suggest to put them to `/opt/build` or to some home subdirectory.

**Comment by zavodnyrichard** [comment]
Still nothing. :/

**Comment by rkfg** [comment]
Then your problem might be somewhere else. Things to try:
- move your library to a non-standard path and build a linux version using CMake and `-DCMAKE_LIBRARY_PATH` so that it wouldn't find it automatically in `/usr` or `/usr/local` and would have to use the path specified
- build MXE from scratch with the changed cmake config; just clone it somewhere else, patch and build as usual, make sure you're using the cmake from that new path, I sometimes accidentally use the old one because I search the shell history with Ctrl+R and forget to change it.

**Comment by zavodnyrichard** [comment]
Well, maybe it's not problem of the MXE CMake, but MXE at all... When I use `i686-w64-mingw32.static-g++ cryptotest.cpp -o cryptotest -lcryptopp`, it can't find library either...

**Comment by rkfg** [comment]
You have to specify the search path with -L flag in that case.

**Comment by zavodnyrichard** [comment]
Thanks for your help. :) Now I solved it but with different solution. I needed modify my CMakeLists.txt, to include directory where lib headers sit. Also I downloaded FindCryptoPP.cmake file from [https://bitbucket.org/sergiu/cryptopp-cmake](https://bitbucket.org/sergiu/cryptopp-cmake) and put it to my source directory and looked for library with **find_package()**, instead of **find_library()**.

My CMakeLists.txt looks like this:
`cmake_minimum_required (VERSION 3.10.0)
project (cryptotest)
add_executable(cryptotest cryptotest.cpp)

#Set location of FindCryptoPP.cmake - source code directory
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}")

#Set CryptoPP root directory
set(CRYPTOPP_ROOT_DIR "/usr/local")

#Include directory, where CryptoPP headers sit
include_directories("/usr/local/include")

#Find CryptoPP library
find_package(CryptoPP REQUIRED)

#Use the CryptoPP library
target_link_libraries(cryptotest ${CryptoPP})`

**NOTE:** If you want use CryptoPP in your project, you need re-compile CryptoPP library as decribed [here](https://stackoverflow.com/a/47140618/5688398), by deactivating use of C++11 support while compilation.
